File: LayoutTests/http/tests/security/contentSecurityPolicy/1.1/resources/scripthash-in-enforced-policy-and-not-in-report-only.php
<?
    header("Content-Security-Policy: script-src 'sha256-c8f8z1SC90Yj05k41+FT0HF/rrGJP94TPLhRvGGE8eM='");
    header("Content-Security-Policy-Report-Only: script-src 'none'");
?>
<!DOCTYPE html>
<html>
<body>
<p id="script-with-hash-result">FAIL did not execute script with hash.</p>
<p id="script-without-hash-result">PASS did not execute script without hash.</p>
<script>document.getElementById("script-with-hash-result").textContent = "PASS did execute script with hash.";</script> <!-- sha256-c8f8z1SC90Yj05k41+FT0HF/rrGJP94TPLhRvGGE8eM= -->
<script>document.getElementById("script-without-hash-result").textContent = "FAIL did execute script without hash.";</script>
</body>
</html>

File: LayoutTests/http/tests/security/contentSecurityPolicy/1.1/resources/scripthash-in-enforced-policy-and-not-in-report-only.py
#!/usr/bin/env python3

import sys

sys.stdout.write(
    'Content-Security-Policy: script-src \'sha256-c8f8z1SC90Yj05k41+FT0HF/rrGJP94TPLhRvGGE8eM=\'\r\n'
    'Content-Security-Policy-Report-Only: script-src \'none\'\r\n'
    'Content-Type: text/html\r\n\r\n'
    '<!DOCTYPE html>\n'
    '<html>\n'
    '<body>\n'
    '<p id="script-with-hash-result">FAIL did not execute script with hash.</p>\n'
    '<p id="script-without-hash-result">PASS did not execute script without hash.</p>\n'
    '<script>document.getElementById("script-with-hash-result").textContent = "PASS did execute script with hash.";</script> <!-- sha256-c8f8z1SC90Yj05k41+FT0HF/rrGJP94TPLhRvGGE8eM= -->\n'
    '<script>document.getElementById("script-without-hash-result").textContent = "FAIL did execute script without hash.";</script>\n'
    '</body>\n'
    '</html>\n'
)